plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -69,6 +80,18 @@ public void done(IJobChangeEvent event) {
     deploy.schedule();
   }
 
+  private IPath createWorkDirectory() throws IOException {
+    String now = Long.toString(System.currentTimeMillis());
+    IPath workDirectory = getTempDir().append(now);
+    Files.createDirectories(workDirectory.toFile().toPath(), new FileAttribute<?>[0]);
+    return workDirectory;
+  }
+
+  private void ensureLoggedInAndSaveCredential(IPath workDirectory, IShellProvider shellProvider) throws IOException, 
```

Comment:
this name is weird. Maybe just logInAndSaveCredential? better yet, how about just logIn? I think the saving of credential is understood